e2etest/bgptests/bgp.go

```diff
@@ -188,16 +188,25 @@ var _ = ginkgo.Describe("BGP", func() {
 			framework.ExpectNoError(err)
 		})
 
+		ginkgo.AfterEach(func() {
+			// Clean previous configuration.
+			err := ConfigUpdater.Clean()
+			framework.ExpectNoError(err)
+		})
+
 		table.DescribeTable("should be exposed by the controller", func(ipFamily ipfamily.Family, poolAddress string, addressTotal int) {
 			poolName := "bgp-test"
 
 			var peerAddrs []string
+			peerAddrToName := make(map[string]string)
```

Comment:
if you keep the map, you can iterate over it and remove the slice.

Reply:
yes, done.